acpreprocessing/stitching_modules/nglink/update_state.py

```diff
@@ -33,6 +38,17 @@ def update_positions(statejson, stitchoutjson, n_pos, factor):
             print("Something went wrong with the stitching output!")
 
 
+def update_positions_consolidated(statejson, stitchoutjson, n_pos, factor):
+    for pos in range(0, n_pos):
+        # print(pos)
+        try:
+            statejson['layers'][0]['source'][pos]['transform']['matrix'][0][3] = stitchoutjson[pos]['position'][0]*factor
+            statejson['layers'][0]['source'][pos]['transform']['matrix'][1][3] = stitchoutjson[pos]['position'][1]*factor
+            statejson['layers'][0]['source'][pos]['transform']['matrix'][2][3] = stitchoutjson[pos]['position'][2]*factor
+        except IndexError:
+            print("Something went wrong with the stitching output!")
```

Comment:
Is it okay to raise this exception and if you need to ignore it in e.g. runmodules you could catch it there?

Another option is to make a custom exception class for these expected errors and catch that.